parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/SnappyCompressor.java

```diff
@@ -96,21 +100,40 @@ public synchronized void setInput(byte[] buffer, int off, int len) {
         "Output buffer should be empty. Caller must call compress()");
 
     if (inputBuffer.capacity() - inputBuffer.position() < len) {
-      ByteBuffer tmp = ByteBuffer.allocateDirect(inputBuffer.position() + len);
-      inputBuffer.rewind();
-      tmp.put(inputBuffer);
-      ByteBuffer oldBuffer = inputBuffer;
-      inputBuffer = tmp;
-      CleanUtil.cleanDirectBuffer(oldBuffer);
-    } else {
-      inputBuffer.limit(inputBuffer.position() + len);
+      resizeInputBuffer(inputBuffer.position() + len);
     }
 
+    inputBuffer.limit(inputBuffer.position() + len);
```

Comment:
The original code doesn't call limit if (inputBuffer.capacity() - inputBuffer.position() < len)  is true